zadania/Program.cs

```diff
@@ -103,6 +103,61 @@ void summary(List<Tuple<int,int,int,int>> links)
     foreach (KeyValuePair<string,int> kvp in summary)
         Console.WriteLine("Podróż {0} odbyła się {1} razy", kvp.Key, kvp.Value);
 }
+int GetAllTo(int i, int j, List<Tuple<int,int,int,int>> links)
+{
+    IEnumerable<Tuple<int,int,int,int>> appropiateLinksList =
+                    from link in links
+                    where link.Item3 == i && link.Item4==j
+                    select link;
+    return appropiateLinksList.Count();
+}
+int GetAllFrom(int i, int j, List<Tuple<int, int, int, int>> links)
+{
+    IEnumerable<Tuple<int, int, int, int>> appropiateLinksList =
+                    from link in links
+                    where link.Item1 == i && link.Item2 == j
+                    select link;
+    return appropiateLinksList.Count();
+}
+void maxTo(List<Tuple<int,int,int,int>> links)
+{
+    int[] result = { 0, 0, 0 };
+    for(int i = 0; i<height; i++)
+    {
+        for(int j = 0; j<width; j++)
+        {
+            int resultAmount = GetAllTo(i, j, links);
+            if (resultAmount > result[0])
+            {
+                result[0]=resultAmount;
+                result[1]=i; result[2]=j;
+            }
+        }
+    }
```

Comment:
Jeżeli ty przechodzisz przez wszystkie możliwe koordynaty mapy żeby sprawdzić który hub cieszy się największym zainteresowaniem to zabijasz wydajność. Dla przykładu masz mapę 20x20, losujesz 5 hubów i 1000 połączeń. Czyli wykonasz 400 operacji na liście 1000 elementowej. Wykorzystaj `Dictionary<Tuple<int, int>, int>` aby obciąć ilość operacji.
Dla każdego elementu w liście links:
 1. Sprawdzasz czy jest w Dictionary => jeśli nie to dodajesz go do niego
 2. Podnosisz o 1 wartość w Dictionary dla klucza Tuple<int, int> - twojego huba